### PR TITLE
Port daml repl to LF2

### PR DESCRIPTION
**Pull request by paulbrauner-da**
`ReplServiceMain.scala` now accepts a major LF version flag that defaults to 1. The service only accepts packages whose major version matches the flag.

The haskell client retrieves the major version from the value of the  `--target` flag passed to `daml repl`.

The repl tests now test both LF1 and LF2.

### CODE REVIEW

File: compiler/damlc/tests/BUILD.bazel [hunk 1]
```diff
@@ -736,15 +758,18 @@ da_haskell_test(
...
-        ":repl-test.dar",
-        ":repl-test-two.dar",
-        ghc_pkg,
-        "//compiler/damlc/pkg-db",
-        "//compiler/damlc/stable-packages",
-        "//compiler/repl-service/server:repl_service_jar",
-        "//daml-script/daml:daml-script.dar",
-    ] + (["@sysctl_nix//:bin/sysctl"] if is_darwin else []),
+    data =
+        [f for major in LF_MAJOR_VERSIONS for f in [
```
Comment (akrmn) [inline compiler/damlc/tests/BUILD.bazel:762]: python!
later we should make these into per-version tests but it's fine for now

File: compiler/repl-service/server/src/main/scala/com/digitalasset/daml/lf/speedy/ReplServiceMain.scala [hunk 2]
```diff
@@ -231,12 +243,28 @@ class ReplService(
...
+    .Default(majorLanguageVersion)
+    .copy(stacktracing = Compiler.FullStackTrace)
+
+  private[this] def validatePackagesLanguageVersion(pkgMap: Map[PackageId, Package]): Unit = {
+    val invalidPackages =
+      pkgMap.view.mapValues(_.languageVersion.major).filter(_._2 != majorLanguageVersion)
+    if (invalidPackages.nonEmpty) {
+      throw new IllegalArgumentException(
+        "the following packages don't have expected major LF version "
+          + s"${majorLanguageVersion}: ${invalidPackages}"
```
Comment (akrmn) [inline compiler/repl-service/server/src/main/scala/com/digitalasset/daml/lf/speedy/ReplServiceMain.scala:256]: we shouldn't print the contents of the packages! the pid on its own would be enough but ideally we would show the package name + version from `package.metadata` if present

Comment (paulbrauner-da) [reply compiler/repl-service/server/src/main/scala/com/digitalasset/daml/lf/speedy/ReplServiceMain.scala:256]: I actually only print the major versions: `mapValues(_.languageVersion.major)`.

File: compiler/damlc/tests/BUILD.bazel (hunk 1 above)
Comment (paulbrauner-da) [reply compiler/damlc/tests/BUILD.bazel:762]: Yeah, I couldn't find a better way of doing this with the primitives exposed by bazel.

File: daml-script/daml3/BUILD.bazel [hunk 3]
```diff
@@ -45,6 +45,17 @@ EOF
...
 ]
 
+# For convenience, we define daml3-script.dar which is always compiled to the
+# default stable version of LF2.
+genrule(
+    name = "daml3-script",
+    # TODO(#17366): use lf_version_default or equivalent once available
+    srcs = [":daml3-script-{}.dar".format(lf_version_latest.get("2"))],
+    outs = ["daml3-script.dar"],
+    cmd = "cp -L $(location :daml3-script-{}.dar) $$PWD/$@".format(lf_version_latest.get("2")),
```
Comment (akrmn) [inline daml-script/daml3/BUILD.bazel:55]: I was going to suggest an `alias` but then I realized that wouldn't change the name of the dar. Oh well, minor bazel annoyance.

File: libs-haskell/test-utils/DA/Test/Sandbox.hs [hunk 4]
```diff
@@ -167,6 +169,8 @@ getCantonConfig conf@SandboxConfig{..} portFile mCerts (ledgerPort, adminPort, d
                 , [ "clock" Aeson..= Aeson.object
                         [ "type" Aeson..= ("sim-clock" :: T.Text) ]
                   | Static <- [timeMode] ]
+                , [ "dev-version-support" Aeson..= devVersionSupport]
+                , [ "non-standard-config" Aeson..= True]
```
Comment (akrmn) [inline libs-haskell/test-utils/DA/Test/Sandbox.hs:173]: does this one always need to be `true`? it feels a bit off

**Review by akrmn — APPROVED** [review]
LGTM, thank you @paulbrauner-da!

File: compiler/damlc/tests/BUILD.bazel (hunk 1 above)
Comment (akrmn) [reply compiler/damlc/tests/BUILD.bazel:762]: yes sorry to be clear I was complaining about bazel/starlark/python

File: compiler/repl-service/server/src/main/scala/com/digitalasset/daml/lf/speedy/ReplServiceMain.scala (hunk 2 above)
Comment (akrmn) [reply compiler/repl-service/server/src/main/scala/com/digitalasset/daml/lf/speedy/ReplServiceMain.scala:256]: 🤦 ah my bad I misread

File: libs-haskell/test-utils/DA/Test/Sandbox.hs (hunk 4 above)
Comment (paulbrauner-da) [reply libs-haskell/test-utils/DA/Test/Sandbox.hs]: Good catch! Done.

**Comment by paulbrauner-da** [comment]
Context: https://github.com/digital-asset/daml/issues/17366

**Pull request merged**